### DISCUSSION TITLE
Camera switch issues and black screen

### DISCUSSION THREAD

**Discussion by yogeshJain0727** [ant-media/Ant-Media-Server#5817] [General]
We are facing some issues related to the camera device list when we try to get it. The problem is that when we try to get the video source list it sometimes gives us half the device (Web Chrome, Android Chrome and on iPhone Safari and Chrome) or sometimes empty data in the object (on iPhone Safari and Chrome) returns. We are also facing camera switch issues during live demo that is when I do not select camera from camera list it automatically opens back camera for streaming or when I select any camera option it opens back camera by default in Android but it shows up black Screen in iOS and sometimes even in Android devices(specially samsung and motorola) but streaming is publishing in background with audio but sometime video got stuck & sometime its publishing with back camera on.

**Comment by lastpeony** [comment]
If you're unable to retrieve the complete device list in the browser, this issue may be connected to either the browser or the system, as it is not directly associated with the Ant Media SDK.

Regarding the black screen problem, could you please provide more details? Are you experiencing difficulties when switching the camera from the front to the back on Android devices? I believe i couldnt get the problem.

**Reply by swatisinha-mis** [reply]
We are facing problem when we start publishing live stream using front camera it shows up black Screen in Android devices(specially samsung & motorola and vivo) in any browser but streaming is publishing in background with audio and camera on. Or when we start publishing live stream using back camera the video gets stuck & sometimes it gets published with back camera. In both scenarios camera switch functionality is not working at all, this causes the live video stream to freeze.

**Reply by lastpeony** [reply]
Okay so you are streaming **through browser** on mobile and facing problems while switching camera from back to front and front to back am i right?

**Reply by yogeshJain0727** [reply]
yes @lastpeony we also create a ticket for that issue "https://github.com/ant-media/Ant-Media-Server/issues/5816"
